cloud-minecraft/cloud-paper/src/main/java/cloud/commandframework/paper/parsers/KeyedWorldArgument.java

```diff
@@ -0,0 +1,151 @@
+    }
+
+    /**
+     * Create a new builder
+     *
+     * @param name Name of the argument
+     * @param <C>  Command sender type
+     * @return Created builder
+     */
+    public static <C> CommandArgument.@NonNull Builder<C, World> newBuilder(final @NonNull String name) {
+        return new KeyedWorldArgument.Builder<>(name);
+    }
+
+    /**
+     * Create a new required argument
+     *
+     * @param name Argument name
+     * @param <C>  Command sender type
+     * @return Created argument
+     */
+    public static <C> @NonNull CommandArgument<C, World> of(final @NonNull String name) {
+        return KeyedWorldArgument.<C>newBuilder(name).asRequired().build();
+    }
+
+    /**
+     * Create a new optional argument
+     *
+     * @param name Argument name
+     * @param <C>  Command sender type
+     * @return Created argument
+     */
+    public static <C> @NonNull CommandArgument<C, World> optional(final @NonNull String name) {
+        return KeyedWorldArgument.<C>newBuilder(name).asOptional().build();
+    }
+
+    /**
+     * Create a new optional argument with a default value
+     *
+     * @param name         Argument name
+     * @param defaultValue Default value
+     * @param <C>          Command sender type
+     * @return Created argument
+     */
+    public static <C> @NonNull CommandArgument<C, World> optional(
+            final @NonNull String name,
+            final @NonNull String defaultValue
+    ) {
+        return KeyedWorldArgument.<C>newBuilder(name).asOptionalWithDefault(defaultValue).build();
+    }
+
+    public static final class Builder<C> extends CommandArgument.Builder<C, World> {
+
+        private Builder(final @NonNull String name) {
+            super(World.class, name);
+        }
+
+        @Override
+        public @NonNull CommandArgument<@NonNull C, @NonNull World> build() {
+            return new KeyedWorldArgument<>(
+                    this.isRequired(),
+                    this.getName(),
+                    this.getDefaultValue(),
+                    this.getSuggestionsProvider(),
+                    this.getDefaultDescription()
+            );
+        }
+
+    }
+
+    public static final class KeyedWorldParser<C> implements ArgumentParser<C, World> {
+
+        @Override
+        public @NonNull ArgumentParseResult<@NonNull World> parse(
+                @NonNull final CommandContext<@NonNull C> commandContext,
+                @NonNull final Queue<@NonNull String> inputQueue
+        ) {
+            final String input = inputQueue.peek();
+            if (input == null) {
+                return ArgumentParseResult.failure(new NoInputProvidedException(
+                        KeyedWorldParser.class,
+                        commandContext
+                ));
+            }
+
+            final NamespacedKey key = NamespacedKey.fromString(input);
```

Comment:
This can throw if the key is invalid iirc, need to catch that and return a failure